x/staking/types/types.proto

```diff
@@ -0,0 +1,273 @@
+syntax = "proto3";
+package cosmos_sdk.x.staking.v1;
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "third_party/proto/tendermint/abci/types/types.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "types/types.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/x/staking/types";
+
+// HistoricalInfoProto contains the historical information that gets stored at
+// each height.
+message HistoricalInfoProto {
+  option (gogoproto.equal) = true;
+
+  tendermint.abci.types.Header header = 1 [ (gogoproto.nullable) = false ];
+  repeated ValidatorProto valset = 2 [ (gogoproto.nullable) = false ];
+}
+
+// CommissionRates defines the initial commission rates to be used for creating
+// a validator.
+message CommissionRates {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  string rate = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string max_rate = 2 [
+    (gogoproto.moretags) = "yaml:\"max_rate\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string max_change_rate = 3 [
+    (gogoproto.moretags) = "yaml:\"max_change_rate\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// Commission defines a commission parameters for a given validator.
+message Commission {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  CommissionRates commission_rates = 1
+      [ (gogoproto.embed) = true, (gogoproto.nullable) = false ];
+  google.protobuf.Timestamp update_time = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true,
+    (gogoproto.moretags) = "yaml:\"update_time\""
+  ];
+}
+
+// Description defines a validator description.
+message Description {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  string moniker = 1;
+  string identity = 2;
+  string website = 3;
+  string security_contact = 4
+      [ (gogoproto.moretags) = "yaml:\"security_contact\"" ];
+  string details = 5;
+}
+
+// ValidatorProto defines the Protocol buffer equivalent of the Validator type.
+message ValidatorProto {
+  option (gogoproto.equal) = true;
+
+  bytes operator_address = 1 [
+    (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.ValAddress",
+    (gogoproto.moretags) = "yaml:\"operator_address\""
+  ];
+  // TODO: We can remove cosmos_sdk.v1.PublicKey in favor of the primitive bytes
+  // once Tendermint supports pubkeys as slices.
+  cosmos_sdk.v1.PublicKey consensus_pubkey = 2 [
+    (gogoproto.moretags) = "yaml:\"consensus_pubkey\"",
+    (gogoproto.nullable) = false
+  ];
+  bool jailed = 3;
+  bytes status = 4;
```

Comment:
this may be better suited as an enum? or at least int32 instead of bytes https://stackoverflow.com/a/13434932/11716895

Reply:
Thanks for the tip!